### DISCUSSION TITLE
What Best way to add sample for esp-idf-hal without loosing the context, cargo, documenation

### DISCUSSION THREAD

**Discussion by joeatbayes** [esp-rs/awesome-esp-rust#40] [Ideas]
I have a sample app where I had to pull together several things that were not clearly shown in the existing samples.  https://github.com/joeatbayes/embedded-rust-examples/tree/main/ex-gpio-measure-time-to-change-high-to-low   The problem is the code for this is a lot more useful if you keep the readme file and images to go along with it.    Is there a way to add all of them as a unit to [esp-idf-hal examples](https://github.com/esp-rs/esp-hal/tree/main/examples/src/bin) directory.  It looks from the git repo that only the rust source is there.   If we cannot put them there then where is a better place to put them?

Note this sample shows the following:
    - Measure elapsed time
    - Measure variable capacitance
    - Detect change in GPIO state
    - Demonstrate Open Drain high-impedance pins 
    - Drive I/O pin high 
    - Delay for a specific duration
    - Read GPIO pin status
    - Change OpenDrive GPIO pin mode between Drain and high impedance
    - Oversampling to improve reading stability

**Comment by bjoernQ** [comment]
I guess you opened this in the wrong repo - this is the bare-metal HAL.

One option might be to keep your repo as is and add it to: https://github.com/esp-rs/awesome-esp-rust

**Reply by joeatbayes** [reply]
Sorry about that.  I must have switched repo accidentally.  I will learn the awesome-esp-rust process for submissions.   I would like to see some of this make it into the esp-idf-hal examples and hal-examples because that is always where I look first for samples.

**Comment by joeatbayes** [comment]
I also have a version of the capacitance sensor that only requires 1 pin.  It was tough to get working in Rust because of some default config settings.  Once I got the default config settings to turn off logging every time pin direction changes it started working just as designed.    The problem with the rust ecosystem is that information was incredibly difficult to find.  So bad that I had given up and gone back to C.   That is an area where embedded rust must improve or it will remain a dabbling experience.    The new version is https://github.com/joeatbayes/embedded-rust-examples/tree/main/ex-gpio-change-pin-direction-var-cap-read-1-pin   It again shows a few things that were not easily found in other examples like using the changing pin direction in a real use case.
